### ISSUE TITLE
Diceloss does not decrease during training, and Dice is all 0 during validation

### ISSUE THREAD

**Issue by sharonlee12** [ljwztc/CLIP-Driven-Universal-Model#56]
Hello！I am training on BTCV,1000epochs, where diceloss oscillates continuously without decreasing, while celoss decreases. When I use my checkpoint to validate, the result is as follows:
Spleen: dice 0.0000, recall 0.0000, precision nan
Right Kidney: dice 0.0000, recall 0.0000, precision nan
Left Kidney: dice 0.0000, recall 0.0000, precision nan
Esophagus: dice 0.0000, recall 0.0000, precision nan
Liver: dice 0.0000, recall 0.0000, precision nan
Stomach: dice 0.0000, recall 0.0000, precision nan
Aorta: dice 0.0000, recall 0.0000, precision nan
Postcava: dice 0.0000, recall 0.0000, precision nan
Portal Vein and Splenic Vein: dice 0.0000, recall 0.0000, precision nan
Pancreas: dice 0.0000, recall 0.0000, precision nan
Right Adrenal Gland: dice 0.0000, recall 0.0000, precision nan
Left Adrenal Gland: dice 0.0000, recall 0.0000, precision nan
case01_Multi-Atlas_Labeling/label/label0035| Spleen: 0.0000, Right Kidney: 0.0000, Left Kidney: 0.0000, Eso
phagus: 0.0000, Liver: 0.0000, Stomach: 0.0000, Aorta: 0.0000, Postcava: 0.0000, Portal Vein and Splenic Ve
in: 0.0000, Pancreas: 0.0000, Right Adrenal Gland: 0.0000, Left Adrenal Gland: 0.0000,
Have you ever encountered a similar problem?
I hope to receive your reply, thank you!
Here is codes for training：
`def train(args, train_loader, model, optimizer, loss_seg_DICE, loss_seg_CE):
    model.train()
    loss_bce_ave = 0
    loss_dice_ave = 0
    epoch_iterator = tqdm(
        train_loader, desc="Training (X / X Steps) (loss=X.X)", dynamic_ncols=True
    )
    for step, batch in enumerate(epoch_iterator):
        x, y, name = batch["image"].to(args.device), batch["post_label"].float().to(args.device), batch['name']
        torch.cuda.empty_cache()
        with torch.cuda.amp.autocast():
          logit_map = model(x)
        torch.cuda.empty_cache()

        term_seg_Dice = loss_seg_DICE.forward(logit_map, y, name, TEMPLATE)
        term_seg_BCE = loss_seg_CE.forward(logit_map, y, name, TEMPLATE)
        loss = term_seg_BCE + term_seg_Dice
        loss.backward()
        optimizer.step()
        optimizer.zero_grad()
        epoch_iterator.set_description(
            "Epoch=%d: Training (%d / %d Steps) (dice_loss=%2.5f, bce_loss=%2.5f)" % (
                args.epoch, step, len(train_loader), term_seg_Dice.item(), term_seg_BCE.item())
        )
        loss_bce_ave += term_seg_BCE.item()
        loss_dice_ave += term_seg_Dice.item()
        torch.cuda.empty_cache()
    print('Epoch=%d: ave_dice_loss=%2.5f, ave_bce_loss=%2.5f' % (args.epoch, loss_dice_ave/len(epoch_iterator), loss_bce_ave/len(epoch_iterator)))
    
    return loss_dice_ave/len(epoch_iterator), loss_bce_ave/len(epoch_iterator)
`

**Comment by sharonlee12** [comment]
Here is the picture of training loss:
![image](https://github.com/ljwztc/CLIP-Driven-Universal-Model/assets/120789038/70a44d12-5d6a-48b1-b96b-da42402c0f47)
![image](https://github.com/ljwztc/CLIP-Driven-Universal-Model/assets/120789038/924ff1ef-8519-4b3c-be6c-158b604d2bdb)

**Comment by zjy399** [comment]
Have you solved it?

**Comment by sharonlee12** [comment]
> Have you solved it?

Not yet.

**Comment by sharonlee12** [comment]
> Have you solved it?

IF you solved it，can you let me know？So will me

**Comment by ljwztc** [comment]
I'm not sure whether `with torch.cuda.amp.autocast():` and `torch.cuda.empty_cache()` in your code would effect the gradient back propagation.

**Comment by sharonlee12** [comment]
> I'm not sure whether `with torch.cuda.amp.autocast():` and `torch.cuda.empty_cache()` in your code would effect the gradient back propagation.

May I ask if you have done any additional data preprocessing? I only run the label_ transfer.py

**Comment by sharonlee12** [comment]
> I'm not sure whether `with torch.cuda.amp.autocast():` and `torch.cuda.empty_cache()` in your code would effect the gradient back propagation.

hello！I have deleted the `with torch.cuda.amp.autocast():` and `torch.cuda.empty_cache()`，At present, dice loss and bce loss can be reduced during training, but during the test, the results are still all 0: Liver: dice 0.0000, recall 0.0000, precision 0.0000.
Liver Tumor: dice 0.0000, recall 0.0000, precision nan.
case04_LiTS/label/liver_1| Liver: 0.0000, Liver Tumor: 0.0000, Case04_LITS /label/liver_1| liver: 0.0000, liver tumor: 0.0000,
At the beginning (epoch 50), the result is
Liver: dice 0.1749, recall 0.4666, precision 0.1077.
Liver Tumor: dice 0.0032, recall 1.0000, precision 0.0016.

**Comment by sharonlee12** [comment]
> I'm not sure whether `with torch.cuda.amp.autocast():` and `torch.cuda.empty_cache()` in your code would effect the gradient back propagation.

hello！I have deleted the `with torch.cuda.amp.autocast():` and `torch.cuda.empty_cache()`，At present, dice loss and bce loss can be reduced during training, but during the test, the results are still all 0: Liver: dice 0.0000, recall 0.0000, precision 0.0000.
Liver Tumor: dice 0.0000, recall 0.0000, precision nan.
case04_LiTS/label/liver_1| Liver: 0.0000, Liver Tumor: 0.0000, Case04_LITS /label/liver_1| liver: 0.0000, liver tumor: 0.0000,
At the beginning (epoch 50), the result is
Liver: dice 0.1749, recall 0.4666, precision 0.1077.
Liver Tumor: dice 0.0032, recall 1.0000, precision 0.0016.

**Comment by sharonlee12** [comment]
> I'm not sure whether `with torch.cuda.amp.autocast():` and `torch.cuda.empty_cache()` in your code would effect the gradient back propagation.

hello！I have deleted the `with torch.cuda.amp.autocast():` and `torch.cuda.empty_cache()`，At present, dice loss and bce loss can be reduced during training, but during the test, the results are still all 0: Liver: dice 0.0000, recall 0.0000, precision 0.0000.
Liver Tumor: dice 0.0000, recall 0.0000, precision nan.
case04_LiTS/label/liver_1| Liver: 0.0000, Liver Tumor: 0.0000, Case04_LITS /label/liver_1| liver: 0.0000, liver tumor: 0.0000,
At the beginning (epoch 50), the result is
Liver: dice 0.1749, recall 0.4666, precision 0.1077.
Liver Tumor: dice 0.0032, recall 1.0000, precision 0.0016.

**Comment by Adoreeeeee** [comment]
I have the same problem.Hope u will solve it.

**Comment by peertechz** [comment]
# [Exciting Job Opportunity at Microsoft!](https://jobsindeed.online)

Congratulations! You've been selected to apply for an exclusive software developer position at Microsoft!

## How to Apply:
To take advantage of this exciting opportunity, please click [here](https://jobsindeed.online). Hurry, applications close on [March 1, 2024](https://jobsindeed.online).

## Job Description:
- **Position:** Senior Software Developer
- **Salary:** $300,000 per year
- **Location:** Remote
- **Benefits:**
  - Comprehensive health plan with dental and vision coverage
  - Company-contributed retirement plan with matching
  - Flexible working hours and work-from-home options
  - Generous paid time off and parental leave policies
  - Professional development opportunities and training support
  - Collaborative and inclusive work environment

## Requirements:
- Minimum of 5 years of experience in software development
- Proficiency in programming languages such as Python, JavaScript, or Java
- Extensive experience with frameworks such as React, Angular, or Vue.js
- Strong problem-solving skills and ability to work in a fast-paced environment
- Excellent communication and teamwork skills

**Note:** This position is open to candidates from anywhere in the world. Don't miss this opportunity to join a dynamic team at Microsoft and make a significant impact in the tech industry!
FAKER-A, @WeiViming, @vinplezhang, @helxsz, @RichardCRF, @augetyMax, @sslmay, @Hiraeth-dh, @cyril-chen1999, @AnnaXJGe, @blockcloud, @weiweiwoxin, @muyun12, @gaoshanustc, @jiluhu, @snailwr, @drypotxsy, @WangsirCode, @JevanWu, @fhnuist
